### ISSUE TITLE
Integration testing resource starts up / shuts down per class. I want one to start up once, clear db after each test, and shut down only on JVM exit

### ISSUE THREAD

**Issue by alechenninger** [lightblue-platform/lightblue-client#222]

**Comment by dcrissman** [comment]
The test framework was designed to work in a way that wiped out mongo between suites. This guarantees a clean slate between suites.

**Comment by alechenninger** [comment]
We should be able to guarantee a clean slate without having to fork the VM, start/stop lightblue, start/stop mongo for every test class. It makes builds very slow.

**Comment by paterczm** [comment]
True, dropping collections should be enough in most cases. If we don't start/stop mongo, we should not need to fork VM.

**Comment by dcrissman** [comment]
Here is one possible solution that exists today using the lightblue-maven-plugin [1]. We would essentially need to move the lightblue servers start/stop to the maven pom (which would implicitly do mongo as well). Honestly, not sure that I am a huge fan of this solution, but just throwing it out there as something that exists today.

[1] https://github.com/lightblue-platform/lightblue-maven-plugin/blob/master/src/it/it-lightblue-server/pom.xml

**Comment by dcrissman** [comment]
What you want is to only setup lightblue/mongo once for multiple suites with mongo collections being cleaned up between suites. The problem, as I see it, is really an issue with junit. It is designed to set-up and tear-down per suite, so if you want something to be set-up for multiple suites, then you either need to do something like what I suggested earlier in the pom.xml or create a suite of suites, something like this suggestion [1]. I think both options have unique shortcomings, but either way it is an application implementation decision.

[1] https://stackoverflow.com/questions/6580670/testsuite-setup-in-junit-4

**Comment by alechenninger** [comment]
We can still use a `@Rule` or `@ClassRule` that can clear out collections before tests run and manage the lifecycle of lightblue and mongo statically: start them up on first use, store them statically to maintain that they're started, and then use a shutdown hook to shut them both down. It will be necessary anyway to share the server information with tests, and to make sure any given test can be run first, from IDE or command line, and still start up lightblue and mongo.
